### PR TITLE
Add ARM schema and tests for ManagedServices

### PR DESCRIPTION
**Pull request by rajivnandivada**

### CODE REVIEW

**Comment by rajivnandivada** [comment]
@Tiano2017 @EvgenyAgafonchikov  could one of you please review this PR? We are trying to target 4/3. thanks.

File: schemas/2015-01-01/deploymentTemplate.json [hunk 1]
```diff
@@ -929,7 +929,9 @@
               {
                 "oneOf": [
                   { "$ref": "https://schema.management.azure.com/schemas/2014-04-01/SuccessBricks.ClearDB.json#/resourceDefinitions/databases" },
-                  { "$ref": "https://schema.management.azure.com/schemas/2015-01-01/Sendgrid.Email.json#/resourceDefinitions/accounts" }
+                  { "$ref": "https://schema.management.azure.com/schemas/2015-01-01/Sendgrid.Email.json#/resourceDefinitions/accounts" },
+                  { "$ref": "https://schema.management.azure.com/schemas/2018-06-01-preview/Microsoft.ManagedServices.json#/resourceDefinitions/registrationAssignments" },
+                  { "$ref": "https://schema.management.azure.com/schemas/2018-06-01-preview/Microsoft.ManagedServices.json#/resourceDefinitions/registrationDefinitions" }
```
Comment (TianoMS) [inline schemas/2015-01-01/deploymentTemplate.json]: these should be put in the other group after line 919. here's it's only for external resources.

File: schemas/2014-04-01-preview/deploymentTemplate.json [hunk 2]
```diff
@@ -905,7 +905,9 @@
                   { "$ref": "https://schema.management.azure.com/schemas/2018-06-01/Microsoft.Security.json#resourceDefinitions/advancedThreatProtectionSettings"},
                   { "$ref": "https://schema.management.azure.com/schemas/2018-06-01/Microsoft.Security.json#resourceDefinitions/informationProtectionPolicies"},
                   { "$ref": "https://schema.management.azure.com/schemas/2019-03-01/Microsoft.Insights.json#/resourceDefinitions/actionGroups" },
-                  { "$ref": "#/definitions/resourceGeneric" }
+                  { "$ref": "#/definitions/resourceGeneric" },
+                  { "$ref": "https://schema.management.azure.com/schemas/2018-06-01-preview/Microsoft.ManagedServices.json#/resourceDefinitions/registrationAssignments" },
+                  { "$ref": "https://schema.management.azure.com/schemas/2018-06-01-preview/Microsoft.ManagedServices.json#/resourceDefinitions/registrationDefinitions" }
```
Comment (TianoMS) [inline schemas/2014-04-01-preview/deploymentTemplate.json]: put them above line 908.

File: schemas/2015-01-01/deploymentTemplate.json [hunk 3]
```diff
@@ -916,7 +916,9 @@
                   { "$ref": "https://schema.management.azure.com/schemas/2018-06-01/Microsoft.Security.json#resourceDefinitions/autoProvisioningSettings"},
                   { "$ref": "https://schema.management.azure.com/schemas/2018-06-01/Microsoft.Security.json#resourceDefinitions/advancedThreatProtectionSettings"},
                   { "$ref": "https://schema.management.azure.com/schemas/2018-06-01/Microsoft.Security.json#resourceDefinitions/informationProtectionPolicies"},
-                  { "$ref": "https://schema.management.azure.com/schemas/2019-03-01/Microsoft.Insights.json#/resourceDefinitions/actionGroups" }
+                  { "$ref": "https://schema.management.azure.com/schemas/2019-03-01/Microsoft.Insights.json#/resourceDefinitions/actionGroups" },
+									{ "$ref": "https://schema.management.azure.com/schemas/2018-06-01-preview/Microsoft.ManagedServices.json#/resourceDefinitions/registrationAssignments" },
```
Comment (TianoMS) [inline schemas/2015-01-01/deploymentTemplate.json]: >									 [](start = 0, length = 9)

please use spaces instead of tabs.

**Pull request merged**